released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.9.0/opam

```diff
@@ -0,0 +1,31 @@
+opam-version: "2.0"
+name: "coq-mathcomp-ssreflect"
+version: "1.9.0"
+maintainer: "Mathematical Components <mathcomp-dev@sympa.inria.fr>"
+
+homepage: "https://math-comp.github.io/"
+bug-reports: "https://github.com/math-comp/math-comp/issues"
+dev-repo: "git+https://github.com/math-comp/math-comp.git"
+license: "CeCILL-B"
+
+build: [ make "-C" "mathcomp/ssreflect" "-j" "%{jobs}%" ]
+install: [ make "-C" "mathcomp/ssreflect" "install" ]
+depends: [ "coq" { ((>= "8.7" & < "8.10~") | (= "dev"))} ]
```

Comment:
```suggestion
depends: [ "coq" { (>= "8.7" & < "8.10~") } ]
```
I'm worried about having `dev` here, since it can stop working at any moment, and makes the `released` repository non-self-contained.

Reply:
I'm fine with that but some users asked for the dev thingy. The point is that they wanted to use a development version of coq (for bug fixes I think) but use stable versions of all dependencies (that is, not the MC.dev package). IIRC @RalfJung was one of these. Is your need still actual? @CohenCyril may also recall this detail...

Reply:
But then these users would use `8.9.dev`, not plain `dev` (which is past `8.10`, so not compatible anyway), wouldn't they?

Reply:
To make the package work with (some) dev versions like `8.10.dev` and the beta, I would set:
```
depends: [ "coq" { (>= "8.7" & < "8.11~") } ]
```
But if someone indeed wants use Coq `master` with stable version of dependencies, this would need the `= "dev"`. However, since the great majority of stable dependencies do not include `= "dev"` this seems to me to have limited uses.

Reply:
@silene why that? Sometimes Coq devs ask "how does this look like on master", and then if packages don't do this "or `dev`" thing, it is a huge pain to figure that out even though most of them work.

> I'm worried about having dev here, since it can stop working at any moment, and makes the released repository non-self-contained.

Following that reasoning, we should also exclude the `8.x.dev` versions as they might break. That makes them useless though.

Same for the `dev` version. FWIW, we do the same amongst our own developments:
```
"coq-stdpp" { (= "dev.2019-05-15.0.7b7ee122") | (= "dev") }
```
Without this, it would be extremely painful to even attempt to build something against a "master" version of Iris/std++.

My understanding is that someone installing the `dev` version of a package does not get to complain when that breaks other package's build. But conversely, if no package even allows `dev` versions then they are useless.

Reply:
I am fully confident in @gares ability to judge whether MathComp compiles or not with 8.10. If he put `8.10~`, then I am convinced that the `dev` dependency is pointless. We are not talking about a hypothetical Coq package that will be released next year; we are talking about 8.10 which is about to be released.

Reply:
> Following that reasoning, we should also exclude the 8.x.dev versions as they might break. That makes them useless though.

I think there's a fundamental difference here. When a user adds the `core-dev` repo, the `8.x.dev` packages expand what the bound `>= "8.x" & < "8.x+1"` means. However, the bounds refer to only the stable Coq for someone who only has the `released` repo, and there is a guarantee that the packages build with that stable Coq. Since all guarantees went out the window when adding `core-dev`, it's fine when something breaks with `8.x.dev` Coq.

Here, `dev` is something that occurs explicitly in the `released` repo which there are no guarantees about. This is my main concern. It doesn't seem like anyone is monitoring when `dev` dependencies in `released` break.

Reply:
Wait, so this *definitely does not build* with 8.10+?  Then sure, the dependency is pointless.

Above it sounded more like "might get broken because there is no compatibility for master".

> Coq. Since all guarantees went out the window when adding core-dev, it's fine when something breaks with 8.x.dev Coq.

`dev` also exists only in `core-dev`, just like `8.10.dev`. So guarantees go out the window the same way. I don't understand the distinction you are making here.

Reply:
For the record, we tried MathComp 1.9 with both `8.10+beta1` and `8.10.dev` from a couple of days ago, and it worked fine. That's why I was suggesting `< "8.11~"` above.

Reply:
this should be fixed now